app/actions/websocket/user.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Q} from '@nozbe/watermelondb';
+import {DeviceEventEmitter} from 'react-native';
+
+import {getTeammateNameDisplaySetting} from '@app/helpers/api/preference';
+import {Events, Preferences} from '@constants';
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@constants/database';
+import DatabaseManager from '@database/manager';
+import WebsocketManager from '@init/websocket_manager';
+import {queryActiveServer} from '@queries/app/servers';
+import {queryConfig} from '@queries/servers/system';
+import {queryCurrentUser, queryUserById} from '@queries/servers/user';
+import PreferenceModel from '@typings/database/models/servers/preference';
+import SystemModel from '@typings/database/models/servers/system';
+import {displayUsername} from '@utils/user';
+
+const {SERVER: {PREFERENCE, SYSTEM}} = MM_TABLES;
+export async function handleUserTypingEvent(serverUrl: string, msg: any) {
+    const currentServer = await queryActiveServer(DatabaseManager.appDatabase!.database);
+    if (currentServer?.url === serverUrl) {
+        const database = DatabaseManager.serverDatabases[serverUrl];
+        if (!database) {
+            return;
+        }
+
+        const user = await queryUserById(database.database, msg.data.user_id);
```

Comment:
do you get the WS if the one typing is yourself? and if you don't have the user loaded in the DB, shouldn't we be requesting it ?

Reply:
The typing user is omitted at server level.
Regarding the user, yes, we probably should.